### PR TITLE
Adding reset method to handle generating a new distinct_id and clearing persistence

### PR DESCRIPTION
**Pull request by stefansedich**
I saw this talked about over in issue #67 and there was a solution given, seems to me as it should be a part of the library as it is a common operation we require.

This PR should address that, if I have missed anything I am happy to fix it up but it appears to be working to me.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ If you originally installed Mixpanel via Bower at its previous home ([https://gi
 
 ## Running tests
 - Install development dependencies: `npm install`
-- Start test server: `npm start`
+- Start test server: `npm test`
```
Comment (stefansedich) [inline README.md]: I seemed to have to run `npm test` and not `npm start` like the documentation stated.

**Comment by tdumitrescu** [comment]
Awesome, thanks so much for this patch! On a first glance it looks good; we'll take a closer look and get it released soon.

**Comment by stefansedich** [comment]
@tdumitrescu just seen a case where this breaks down:

1. user logs in and I identify() them
2. user logs out and I reset
3. someone else on this computer signs up and I attempt to alias them: "Attempting to create alias for existing People user - aborting."

Will have a look into it unless you have any ideas, I assume I just need to clear some more flags.

**Comment by stefansedich** [comment]
Morning @tdumitrescu have you had any luck looking into this one further to see it's viability for a merge?

**Comment by stefansedich** [comment]
Small change on this one, realized I should probably not call identify again to set the new distinct_id but probably just call register_once like the init does.

**Comment by stefansedich** [comment]
Any news on this one? been a while now, would love to remove my custom hack.

**Comment by tdumitrescu** [comment]
Sorry about the delay, will be getting back to this one this week!

**Comment by tdumitrescu** [comment]
Added to [rc 2.8.0](https://github.com/mixpanel/mixpanel-js/pull/80); should be released next week.

**Comment by tdumitrescu** [comment]
released in v2.8.0, thanks!

**Pull request merged**